developer-extension/src/background/devtoolsPanelConnection.ts

```diff
@@ -0,0 +1,59 @@
+import type { BackgroundToDevtoolsMessage, DevtoolsToBackgroundMessage } from '../common/types'
+import { EventListeners } from '../common/eventListeners'
+import { createLogger } from '../common/logger'
+
+const logger = createLogger('devtoolsPanelConnection')
+
+const devtoolsConnections = new Map<number, chrome.runtime.Port>()
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** naming suggestion‏
```suggestion
const devtoolsConnectionByTabId = new Map<number, chrome.runtime.Port>()
```